sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -717,12 +717,11 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   /**
    * Return the value of an optional Spark SQL configuration property for the given key. If the key
-   * is not set yet, throw an exception.
+   * is not set yet, returns None.
    */
-  def getConf[T](entry: OptionalConfigEntry[T]): T = {
+  def getOptionalConf[T](entry: OptionalConfigEntry[T]): Option[T] = {
```

Comment:
This method is now only used in the tests you're adding... so it could be removed, and your tests slightly changed, to just use the existing `getConf`.